src/tee_host.rs

```diff
@@ -345,6 +345,150 @@ pub enum TeeHostFunction {
+    ///     smaller than the requested page size (e.g. 4kB if requested size is 2MB).
+    ///   - The pages are physically contiguous and suitably aligned to the requested huge
+    ///     page size.
+    ///   - The pages are all in the same state, either the “Blocked” state and uniquely owned
+    ///     by the TVM, or in the “BlockedShared” state and owned by the host.
+    ///   - The TLB version at which the page was invalidated is older than the TVM’s current
+    ///     TLB version.
+    ///
+    /// After verifying these pre-conditions are met, the TSM:
+    ///   - Promotes the non-leaf PTE mapping the huge page to a leaf PTE with the PFN of the
+    ///     huge page and marks it as present.
+    ///   - Returns the now-unused page-table page to the TVM’s page-table page pool.
+    ///   - Places all the sub-pages in the “Mapped” state (if “Blocked”) or “Shared” state
+    ///     (if “BlockedShared”). If the page was “BlockedShared”, the reference count of the
+    ///     page is restored.
+    ///
+    /// Returns 0 on success.
+    ///
+    /// a6 = 18
+    TvmPromotePage {
+        /// a0 = guest_id   
+        guest_id: u64,
+        /// a1 = guest physical address
+        guest_addr: u64,
+        /// a2 = page size
+        page_type: TsmPageType,
+    },
+    /// Demotes a huge page mapping to a set of contiguous mappings at the target size.
+    ///
+    /// The TSM verifies that:
+    ///   - The page is currently marked invalid in the TVM’s page table and has a size one
+    ///     larger than the requested page size (e.g. 2MB if requested size is 4kB).
+    ///   - The page is in either the “Blocked” state and uniquely owned by the TVM, or in
+    ///     the “BlockedShared” state and owned by the host.
+    ///   - The TLB version at which the page was invalidated is older than the TVM’s current
+    ///     TLB version.
+    ///   - There is a free page in the TVM’s page-table page pool.
+    ///
+    /// After verifying these pre-conditions are met, the TSM:
+    ///   - Allocates and fills in a page-table page mapping the sub-pages of the huge page
+    ///     to be split, marking each PTE as present.
+    ///   - Installs a non-leaf PTE pointing the newly-allocated page-table page in place of
+    ///     the huge-page leaf PTE.
+    ///   - Places all the sub-pages in the “Mapped” state (if “Blocked”) or “Shared” state
+    ///     (if “BlockedShared”). If the page was “BlockedShared”, the reference count of the
+    ///     page is restored.
+    ///
+    /// Returns 0 on success.
+    ///
+    /// a6 = 19
+    TvmDemotePage {
+        /// a0 = guest_id   
+        guest_id: u64,
+        /// a1 = guest physical address
+        guest_addr: u64,
+        /// a2 = page size
+        page_type: TsmPageType,
+    },
+    /// Removes mappings from a TVM. The range to be unmapped must already have been invalidated
+    /// and fenced, and must lie within a removable region of guest physical address space.
```

Comment:
probably need to define what a "removable region of guest physical address space" is.

Reply:
I added a sentence below to cover this.